socorro/external/boto/connection_context.py

```diff
@@ -269,8 +285,8 @@ class S3ConnectionContext(ConnectionContextBase):
         likely_to_be_changed=True,
     )
 
-    def __init__(self, config, quit_check_callback=None):
-        super(S3ConnectionContext, self).__init__(config)
+    def __init__(self, config, namespace='', quit_check_callback=None):
+        super(S3ConnectionContext, self).__init__(config, namespace)
```

Comment:
I'd change this and the other instances like this to:

```
super(S3ConnectionContext, self).__init__(config, namespace=namespace)
```

which makes it a lot easier for us to adjust the signatures later on without arguments ending up in the wrong place.

For example, say we add another argument and for some reason in the API it makes sense to put it after the `config` arg:

```
def __init__(self, config, prefix, namespace='', quit_check_callback=None)
```

Now we'll have to go through and fix all the other situations. If instead we were using keyword-style argument passing, we wouldn't have to do anything further.

Plus it makes the code easier to read and debug. If we're doing keyword-style arguments, then we know the value ends up in the right place. If we're doing index-style arguments, then we have to remember signatures and order and all that.

I don't think we need to pass `config` as a keyword-style argument because we have an unwritten convention in Socorro that `config` is always the first argument in configman-style classes and it's always passed in.